### PR TITLE
This patch ensures 'count' is a valid offset.

### PR DESCRIPTION
**Pull request by SMFloris**
| Q | A |
| --- | --- |
| Branch? | 3.1 for fixes |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #19804 |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

**Comment by iltar** [comment]
Can you add a test for this? You're altering the data structure here

File: src/Symfony/Component/Ldap/LdapClient.php [hunk 1]
```diff
@@ -70,7 +70,8 @@ public function find($dn, $query, $filter = '*')
             $resultEntry = array();
 
             foreach ($entry->getAttributes() as $attribute => $values) {
-                $resultAttribute = $values;
+                $resultAttribute = array();
+                $resultAttribute[] = $values;
```
Comment (iltar) [inline src/Symfony/Component/Ldap/LdapClient.php]: Should be written as 
```
$resultAttribute = array($values);
```

But note that this changes the structure

**Comment by SMFloris** [comment]
Indeed, this does alter the structure. Today I am going to play around with it a bit, as I have to get some attributes/values from LDAP and also write/rewrite the test.

It was a blocker for us, we are using Microsoft's Active Directory and at least we can authenticate properly using the ldap provider.

**Comment by nicolas-grekas** [comment]
ping @csarrazi

**Comment by csarrazi** [comment]
:-1 on changing the structure. We need to investigate and check where the exact issue comes from. Whether it's Active Directory which returns some LDAP attributes directly as string values, and improve the `filterAttributes` method of the `Symfony\Component\Ldap\Adapter\ExtLdap\Collection` class to handle this case.

**Comment by fabpot** [comment]
I agree with @csarrazi. Closing as this cannot be merged as is and we have a referenced ticket explaining the issue. @csarrazi Can you have a look at this issue?

**Comment by csarrazi** [comment]
@fabpot Yes, we are investigating the issue in #19804

**Comment by csarrazi** [comment]
Should be fixed in #19975

**Pull request closed without merging**